### DISCUSSION TITLE
built-in testing trigger

### DISCUSSION THREAD

**Discussion by FranksRobins** [Kludex/uvicorn#1894] [Ideas]
Hi there uvicorn team,
First of all I would like to thank you for making uvicorn uvicorn. While using it for development with Fastapi I found that I could use a testing functionality. Now maybe I have misread the docs but it seems like there is no such thing currently available. I think a exec trigger that would be passed as an argument when launching uvicorn would be helpful. Something like 
```
--with-exec="path/to/command/or/script" 
```
that would be used in pair with the **--reload** argument could help while in development. This specific command (or script) would be run every time the server reloads after a change in the code source. 

A complete example would look like the following:
```
uvicorn my_api:app --reload --with-exec /usr/local/bin/my_api_test
```
That script could be some simple curl requests to test the endpoints or a more elaborated testing unit.

I found that this simple functionality would be a great help when developing locally. If there is already something similar, I would be happy to be aware of it. If I am missing something I would also be happy to be aware of it!

Thanks

**Comment by euri10** [comment]
> I found that this simple functionality would be a great help when developing locally.

While I'm very dubious we should add that feature, I'm still interested in knowing what kind of issue for the developper it would solve ?
